### DISCUSSION TITLE
RoundRobinDestinationConnectionStrategy with connectionTTL

### DISCUSSION THREAD

**Discussion by ronakgpatel** [logfellow/logstash-logback-encoder#1023] [Q&A] [answered]
Hi,

We have configured RoundRobinDestinationConnectionStrategy  as connection strategery for LogstashTcpSocketAppender.
We have configured "connectionTTL" as 30 seconds. 

However, the observation is that if the connection is idle for certain duration (e.g. 30 minutes) then there is TCP_RESET in response to the event. Due to this, the log is not received to LogTransformer component, and it results in loss of log entry.
Subsequence log events are captured and worked as expected. 

Is it important to configure the keepAlive when configured connectionTTL? As per the source, the connectionTTL(close and reopening connection) is done AFTER the event is sent. i.e. If there is no event sent, then there is not re-connection happening hence the connection may be reset.

Please suggest if configuring keepAlive along with RoundRobinDestinationConnectionStrategy+connectionTTL is supported or not.

Thanks.

**Comment by philsttr** [comment, marked as answer]
Yes, set the [keepAliveDuration ](https://github.com/logfellow/logstash-logback-encoder?tab=readme-ov-file#keep-alive) to prevent the idle timeouts from occurring.
